### PR TITLE
Remove unnecessary attention mask

### PR DESCRIPTION
**Pull request by xrennvidia**
# What does this PR do ?

Remove unnecessary attention masks. [Related MCore MR is here.](https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/merge_requests/1259)

### CODE REVIEW

**Comment by xrennvidia** [comment]
jenkins

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -1125,6 +1125,7 @@ def get_batch(self, data_iterator, tuning):
             'tokens': data["tokens"],
             'labels': data["labels"],
             'loss_mask': data["loss_mask"],
+            'attention_mask': None if "attention_mask" not in data else data["attention_mask"],
```
Comment (akoumpa) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:1128]: nit: `getattr(data, "attention_mask", None)

**Review by akoumpa — APPROVED** [review]
Just a minor thing, feel free to ignore if you are satisfied with the current syntax.

LGTM, Thanks!
